### PR TITLE
[E2E] add test case for resource interpreter customization

### PR DESCRIPTION
**Pull request by jameszhangyukun**
Signed-off-by: zhangyukun <38148677+jameszhangyukun@users.noreply.github.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add test case for resource interpreter customization:
- DependencyInterpretation

**Which issue(s) this PR fixes**:
part of #2371 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by jameszhangyukun** [comment]
cc @XiShanYongYe-Chang

File: test/e2e/resourceinterpreter_test.go [hunk 1]
```diff
@@ -608,4 +608,70 @@ end `,
...
+					framework.WaitServiceAccountPresentOnClusterFitWith(targetCluster, deployment.Namespace, sa.GetName(),
+						func(sa *corev1.ServiceAccount) bool {
+							return true
+						})
+				})
+
+				ginkgo.By("make the sa is not referenced by the deployment ", func() {
+					framework.UpdateDeploymentServiceAccountName(kubeClient, deployment, "default")
+					framework.WaitServiceAccountDisappearOnCluster(targetCluster, deployment.GetNamespace(), sa.GetName())
+				})
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:671]: This check can be removed. We just need to make sure that the interpreter executes correctly.

File: test/e2e/resourceinterpreter_test.go [hunk 2]
```diff
@@ -608,4 +608,70 @@ end `,
 			})
 		})
 	})
+
+	ginkgo.Context("InterpreterOperation DependencyInterpretation testing", func() {
+		ginkgo.BeforeEach(func() {
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:613]: Don't need this `BeforeEach`.

**Comment by jameszhangyukun** [comment]
cc @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/approve

**Pull request merged**